src/entries/popup/hooks/swap/useSwapInputs.ts

```diff
@@ -248,6 +253,92 @@ export const useSwapInputs = ({
     [assetToBuyValue, independentField],
   );
 
+  const determineOutputCurrency = useCallback(
+    (asset: ParsedSearchAsset | null) => {
+      if (!asset) return null;
+
+      const { chainId } = asset;
+
+      const supportedChain = SUPPORTED_CHAINS.find(
+        (chain) => chain.id === chainId,
+      );
+
+      if (!supportedChain) return null;
+
+      const isNativeCurrencyEth =
```

Comment:
why just checking against eth only? shouldn't we be checking if is the native currency for all chains? even the ones where eth is not the native asset

Reply:
Yeah agree i want to add native asset. For now i think we'll just do ETH and probably do a followup for all native tokens, e.g Degen asset for Degen chain since it's native.